libs/stats/odc/stats/utils.py

```diff
@@ -170,3 +177,87 @@ def dedup_s2_datasets(dss):
         out.append(chunk[-1])
         skipped.extend(chunk[:-1])
     return out, skipped
+
+
+def fuse_products(type_1: DatasetType, type_2: DatasetType, dc: Datacube):
```

Comment:
```suggestion
def fuse_products(type_1: DatasetType, type_2: DatasetType, dc: Datacube) -> DatasetType:
```

Reply:
Let's get rid of `dc` parameter, see comment below, but add optional `name: Optional[str] = None` specifying name for fused product, current default will do when it's `None`